azure-mgmt/tests/mgmt_testcase.py

```diff
@@ -130,7 +130,9 @@ def _scrub(self, val):
         val = super(AzureMgmtTestCase, self)._scrub(val)
         real_to_fake_dict = {
             self.settings.SUBSCRIPTION_ID: self.fake_settings.SUBSCRIPTION_ID,
-            self.settings.AD_DOMAIN:  self.fake_settings.AD_DOMAIN
+            self.settings.AD_DOMAIN:  self.fake_settings.AD_DOMAIN,
```

Comment:
We should stop doing this, because `AD_DOMAIN` and now `TENANT_ID` and `CLIENT_OID` are optionals. Problems is this will break `mgmt_settings_real.py` everywhere for unecessary parameters (I got the issue when I introduced `AD_DOMAIN` but now there is way more users). What do you think to replace this dict by:
```python
constants_to_scrub = ['SUBSCRIPTION_ID', 'AD_DOMAIN', 'TENANT_ID', 'CLIENT_OID']
real_to_fake_dict = {getattr(self.settings, key): getattr(self.fake_settings, key) for key in constants_to_scrub if hasattr(self.settings, key}
```